### PR TITLE
Avoid calling seekToRow on the ORC reader

### PR DESCRIPTION
**Pull request by nileema**
Calling seekToRow on the ORC reader tries to recompute the offsets each
time which in this implementation can be an expensive operation. Avoid
calling seekToRow instead iterate through the file sequentially.

### CODE REVIEW

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java [hunk 1]
```diff
@@ -682,17 +685,21 @@ public void finishDelete(ConnectorSession session, ConnectorTableHandle tableHan
...
-                    newShards.addAll(delta.getNewShards());
+                    oldShardUuidsBuilder.addAll(delta.getOldShardUuids());
+                    newShardsBuilder.addAll(delta.getNewShards());
                 });
 
-        shardManager.replaceShardUuids(transactionId, tableId, columns, oldShardUuids.build(), newShards.build());
+        Set<UUID> oldShardUuids = oldShardUuidsBuilder.build();
+        List<ShardInfo> newShards = newShardsBuilder.build();
+
+        log.debug("Finishing delete transactionId %s tableId %s shards deleted %s shards rewritten %s", transactionId, tableId, oldShardUuids.size() - newShards.size(), newShards.size());
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java]:
```java
log.info("Finishing delete for tableId %s (removed: %s, rewritten: %s)", tableId, oldShardUuids.size() - newShards.size(), newShards.size());

Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java]: This logging change is unrelated, move to separate commit

File: presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcFileRewriter.java [hunk 2]
```diff
@@ -70,6 +74,9 @@ public static OrcFileInfo rewrite(File input, File output, BitSet rowsToDelete)
                 throw new IOException("File has too many rows");
             }
             int inputRowCount = Ints.checkedCast(reader.getNumberOfRows());
+            long rowsToRewrite = reader.getNumberOfRows() - rowsToDelete.cardinality();
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcFileRewriter.java]: `cardinality()` has to walk the bitset. Introduce a variable

File: presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcFileRewriter.java [hunk 3]
```diff
@@ -78,7 +85,9 @@ public static OrcFileInfo rewrite(File input, File output, BitSet rowsToDelete)
 
             try (Closer<RecordReader, IOException> recordReader = closer(reader.rows(), RecordReader::close);
                     Closer<Writer, IOException> writer = closer(createWriter(path(output), writerOptions), Writer::close)) {
-                return rewrite(recordReader.get(), writer.get(), rowsToDelete, inputRowCount);
+                OrcFileInfo fileInfo = rewrite(recordReader.get(), writer.get(), rowsToDelete, inputRowCount);
+                log.debug("Finished rewriting shard %s in %s input rows %s rewritten rows %s", input.getName(), nanosSince(start).convertToMostSuccinctTimeUnit(), inputRowCount, rowsToRewrite);
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcFileRewriter.java]:
```java
log.debug("Rewrote file %s in %s (input rows: %s, output rows: %s)", input.getName(), nanosSince(start), inputRowCount, outputRowCount);

Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcFileRewriter.java]: `nanosSince` is already succinct

File: presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcFileRewriter.java (hunk 2 above)
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcFileRewriter.java]: Name this `outputRowCount`

Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcFileRewriter.java]: Move this to a separate commit for logging

File: presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcFileRewriter.java [hunk 4]
```diff
@@ -87,28 +96,34 @@ private static OrcFileInfo rewrite(RecordReader reader, Writer writer, BitSet ro
             throws IOException
     {
         Object object = null;
-        int row = 0;
+        int selectedRow = 0;
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcFileRewriter.java]: Keep this as `row`

File: presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcFileRewriter.java [hunk 5]
```diff
@@ -87,28 +96,34 @@ private static OrcFileInfo rewrite(RecordReader reader, Writer writer, BitSet ro
...
         while (true) {
             if (Thread.currentThread().isInterrupted()) {
                 throw new InterruptedIOException();
             }
 
-            row = rowsToDelete.nextClearBit(row);
-            if (row >= inputRowCount) {
+            if (selectedRow >= inputRowCount) {
                 return new OrcFileInfo(rowCount, uncompressedSize);
             }
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcFileRewriter.java]: We can simplify this:
```java
if (row > reader.getRowNumber()) {
    reader.next(object);
    continue;
}

**Comment by nileema** [comment]
@electrum updated

**Comment by electrum** [comment]
Looks good
